tests/integration_tests/performance/test_vsock_ab.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the VSOCK throughput of Firecracker uVMs."""
+
+import os
+
+import pytest
+
+from framework.utils import CpuMap
+from framework.utils_iperf import IPerf3Test, emit_iperf3_metrics
+from framework.utils_vsock import VSOCK_UDS_PATH, make_host_port_path
+
+BASE_PORT = 5201
+
+# How many clients/servers should be spawned per vcpu
+LOAD_FACTOR = 1
+
+# Time (in seconds) for which iperf "warms up"
+WARMUP_SEC = 3
+
+# Time (in seconds) for which iperf runs after warmup is done
+RUNTIME_SEC = 20
+
+# VM guest memory size
+GUEST_MEM_MIB = 1024
+
+
+class VsockIPerf3Test(IPerf3Test):
+    """IPerf3 runner for the vsock throughput performance test"""
+
+    def __init__(self, microvm, mode, payload_length):
+        super().__init__(
+            microvm,
+            BASE_PORT,
+            RUNTIME_SEC,
+            WARMUP_SEC,
+            mode,
+            LOAD_FACTOR * microvm.vcpus_count,
+            2,
+            iperf="/usr/local/bin/iperf3-vsock",
+            payload_length=payload_length,
+        )
+
+    def host_command(self, port_offset):
+        return (
+            super()
+            .host_command(port_offset)
+            .with_arg("--vsock")
+            .with_arg("-B", os.path.join(self._microvm.path, VSOCK_UDS_PATH))
+        )
+
+    def spawn_iperf3_client(self, client_idx):
+        # Bind the UDS in the jailer's root.
+        self._microvm.create_jailed_resource(
+            os.path.join(
+                self._microvm.path,
+                make_host_port_path(VSOCK_UDS_PATH, self._base_port + client_idx),
+            )
+        )
+        # The rootfs does not have iperf3-vsock
+        iperf3_guest = "/tmp/iperf3-vsock"
+
+        self._microvm.ssh.scp_put(self._iperf, iperf3_guest)
+        self._guest_iperf = iperf3_guest
+        return super().spawn_iperf3_client(client_idx)
+
+    def guest_command(self, port_offset):
+        return super().guest_command(port_offset).with_arg("--vsock")
```

Comment:
This looks completely same with the one in test_vsock_throughput.py. Can't we import it?

Reply:
Mh, yeah, this is copied 1:1 from test_vsock_throughput.py. My philosophy with these was that I want the two test implementations to be completely independent, which is why I copied everything over (as test_vsock_throughput.py will hopefully go away soon).